src/pymmcore_widgets/useq_widgets/_positions.py

```diff
@@ -185,16 +198,63 @@ def value(
         ):
             if not r.get(self.NAME.key, True):
                 r.pop(self.NAME.key, None)
-            out.append(useq.Position(**r))
+
+            if not self.use_af.isChecked():
+                pos = useq.Position(**r)
+            else:
+                pos = useq.Position(**r)
+                af_offset = r.get(self.AF.key, None)
+                if pos.sequence is None:
+                    # if there is no sub-sequence, create a new one with the autofocus
+                    pos = pos.replace(
+                        sequence=useq.MDASequence(
+                            autofocus_plan=self._get_autofocus_plan(af_offset)
+                        )
+                    )
+                else:
+                    # if there is a sub-sequence, add the autofocus plan to it
+                    pos = pos.replace(
+                        sequence=pos.sequence.replace(
+                            autofocus_plan=self._get_autofocus_plan(af_offset)
+                        )
+                    )
+
+            out.append(pos)
         return tuple(out)
 
     def setValue(self, value: Sequence[useq.Position]) -> None:  # type: ignore
         """Set the current value of the table."""
         _values = []
+        _use_af = False
+        # _af_z_devices: set[str] = set()
         for v in value:
             if not isinstance(v, useq.Position):  # pragma: no cover
                 raise TypeError(f"Expected useq.Position, got {type(v)}")
-            _values.append(v.model_dump(exclude_unset=True))
+
+            _af = {}
+            if v.sequence is not None and v.sequence.autofocus_plan is not None:
+                # if the sub-sequence is empty, set it to None. Else we simply
+                # exclude the autofocus plan
+                sub_seq_dict = v.sequence.model_dump(
+                    exclude_unset=True, exclude={"autofocus_plan"}
+                )
+                sub_seq = useq.MDASequence(**sub_seq_dict) if sub_seq_dict else None
+
+                # get autofocus plan device name and offset
+                _af_offset = v.sequence.autofocus_plan.autofocus_motor_offset
+
+                # set the autofocus offset that will be added to the table
+                _af = {self.AF.key: _af_offset}
+
+                # remopve autofocus plan from sub-sequence
+                v = v.replace(sequence=sub_seq)
+
+                _use_af = True
```

Comment:
this logic feels a bit long/convoluted. need to look later when I'm able to run it locally